packages/frontend-shared/src/gql-components/HeaderBarContent.cy.tsx

```diff
@@ -162,6 +162,44 @@ describe('<HeaderBarContent />', { viewportWidth: 1000, viewportHeight: 750 }, (
     })
   })
 
+  context('responsive design', () => {
+    beforeEach(() => {
+      cy.mountFragment(HeaderBar_HeaderBarContentFragmentDoc, {
+        onResult: (result) => {
+          result.currentProject = null
+        },
+        render: (gqlVal) => (
+          <div class="border-current border-1 h-700px resize overflow-auto">
+            <HeaderBarContent gql={gqlVal} />
+          </div>
+        ),
+      })
+    })
+
+    // https://github.com/cypress-io/cypress/issues/21842
+    it('shows docs menu correctly on small viewports', () => {
+      // Simulate the small viewport.
+      cy.viewport(767, 800)
+
+      // show docs menu
+      cy.contains('button', text.docsMenu.docsHeading).click()
+
+      // docs menu flex direction is column when viewport width is small
+      cy.findByTestId('docs-menu-container').should('have.css', 'flex-direction', 'column')
```

Comment:
Let's add a `cy.percySnapshot()`for each viewport. I know they won't run for this outside contributor PR but I think it's a better test than asserting the flex-direction. It would be easy to rearrange the internals of this component in a way that breaks one breakpoint (say by adding an intermediate layout container) but still passes this test.

Reply:
@marktnoonan `cy.percySnapshot` comes along free for the ride -- it's in the top `afterEach` :)